client/notices/index.js

```diff
@@ -37,7 +37,12 @@ module.exports = {
 			container: container,
 			button: options.button,
 			href: options.href,
-			onClick: options.onClick,
+			onClick: ( event ) => {
+				if ( 'function' === typeof options.onClick ) {
```

Comment:
Recommended it is, Yoda style with WordPress. [Not so much](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#yoda-conditions), with Calypso.